once/_iterator_wrappers.py

```diff
@@ -142,6 +145,7 @@ def yield_results(self) -> collections.abc.Generator:
                     action = _IteratorAction.YIELDING
                     yield_value = self.results[i]
             if action == _IteratorAction.WAITING:
+                time.sleep(0)
```

Comment:
Definitely need an explanation for this time.sleep(0), I don't understand why it should be necessary.

Reply:
Added a comment, it's the same situation as `await asyncio.sleep(0)` - the code is so much slower without this because the default is to wait 5ms before switching threads. Commenting this line out takes test from 0.9 sec to 5 sec.

Reply:
This is actually different from the async one which is important for correctness (we should change it to a synchronization primitive, but it works).

Sounds good, if you measured it, I'm game. At some later date, we'll rig this up as a threading.cond.